### PR TITLE
use crd manager to update and clean crds

### PR DESCRIPTION
**Pull request by qiujian16**
Signed-off-by: Jian Qiu <jqiu@redhat.com>

### CODE REVIEW

File: pkg/operators/crdmanager/manager.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+func (m *manager[T]) Clean(ctx context.Context) error {
+	var errs []error
+	var remainingCRDs []string
+	for name := range m.crds {
+		_, err := m.client.Get(ctx, name, metav1.GetOptions{})
+		if errors.IsNotFound(err) {
+			continue
+		} else if err == nil {
+			remainingCRDs = append(remainingCRDs, remainingCRDs...)
```
Comment (zhiweiyin318) [inline pkg/operators/crdmanager/manager.go]: append the name into remainingCRDs here?

File: pkg/operators/crdmanager/manager.go [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	versionutil "k8s.io/apimachinery/pkg/version"
+	"k8s.io/klog/v2"
+	"open-cluster-management.io/registration-operator/pkg/version"
+)
+
+const versionLabelKey = "operator.open-cluster-management.io/version"
```
Comment (zhujian7) [inline pkg/operators/crdmanager/manager.go]: What is this for? could add more comments on this?

File: pkg/operators/crdmanager/manager.go [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+		return err
+	}
+
+	// if existingVersion is higher than the required version, do not update crd.
+	accessor, err = meta.Accessor(existing)
+	if err != nil {
+		return err
+	}
+	existingVersion, ok := accessor.GetLabels()[versionLabelKey]
+	requiredVersion := version.Get().GitVersion
```
Comment (zhujian7) [inline pkg/operators/crdmanager/manager.go]: If we are in development phase, the `version.Get().GitVersion` could be like `v0.9.0-16-g889bd8b`. can this value be comparable?

**Pull request merged**